ouroboros-consensus/docs/interface-CHANGELOG.md

```diff
@@ -0,0 +1,117 @@
+# Interface Change Log
+
+- This change log is primarily intended to benefit downstream Cardano
+  development teams. We therefore are not committing to discuss changes that
+  aren't relevant to that audience. If such changes belong anywhere, it would
+  most likely be the `cardano-node` release notes.
+
+- So far, versioning of "internal" Cardano packages is based only on commit
+  hashes, except for the `cardano-node` repo.
+
+For those reasons, we organize this log chronologically: with most recent
+changes near the top.
+
+We recommend three ways to query this log.
+
+- `git diff revOld revNew -- ouroboros-consensus/docs/interface-CHANGELOG.md`
+
+  That command identifies the relevant log entries, those that discuss changes
+  between `revOld` and `revNew`. This is probably what downstream maintainers
+  most often need.
+
+- `git blame [--first-parent] --
+  ouroboros-consensus/docs/interface-CHANGELOG.md`
+
+  That command identifies the commit that introduced each log entry, in case
+  you'd like to learn more about the changes discussed by some entry.
+
+  We make a best effort to add a log entry in the same PR that makes the changes
+  that need to be discussed in this file. Also, the commits on our mainline
+  branch -- which is the only branch we assume downstream teams are using -- are
+  all PR merge commits. We don't do fast-forward merges. Each such merge commit,
+  thanks to `bors`, lists the corresponding PR number. Thus the `--first-parent`
+  flag will tell you which PR to investigate if you're interested in more
+  details/discussion around a particular log entry.
+
+- Manually inspect the dates in the (chronologically sorted) entries. We make a
+  best effort to ensure the entry's date indicates when the corresponding PR was
+  merged. We write "Circa YYYY-MM-DD" to emphasize that, for various reasons,
+  the date might not be exact (time zones, pathologically slow CI runs, etc). If
+  you do need precise timestamps, see instead the `git blame` output.
+
+Inevitably, we will eventually need to add errata to a pre-existing log entry.
+In order for the recommended `diff` command to be minimal and the recommended
+`blame` to be most legible, we make a best effort to avoid editing the lines
+within old entries when updating this file. Those lines are ideally immutable,
+just like their corresponding commits. If we find a particularly problematic
+mistake in an old entry, we address it by adding new lines at the top of the log
+entry. The new list item within the old entry should follow this template `-
+Errata Circa 2022-12-30: The original discussion, still found below,
+oversimplifies the ...`. Such errata only correct mistakes in the original log
```

Comment:
```suggestion
oversimplifies the ...`. Such errata only corrects mistakes in the original log
```

Reply:
`errata` is plural, so `correct` is ... correct. Yeah?

Reply:
Oh, I see. Sorry about that.